vsphere/resource_vsphere_host_virtual_switch.go

```diff
@@ -0,0 +1,135 @@
+package vsphere
+
+import (
+	"fmt"
+	"time"
+
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+)
+
+func resourceVSphereHostVirtualSwitch() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the virtual switch.",
+			Required:    true,
+			ForceNew:    true,
+		},
+		"host": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the host to put this virtual switch on. This is ignored if connecting directly to ESXi, but required if not.",
```

Comment:
The problem here is that our provider targets ESXi and vCenter - so we need to accommodate both cases. With ESXi this option is superfluous because there is only one host and datacenter the provider can target - the host that the provider is connected to, and the "default" datacenter.

Hence, I would rather have an error trigger when someone tries to use this with vCenter without these options specified, versus have someone enter dummy data when working with ESXi:

Good: 

```hcl
resource "vsphere_host_virtual_switch" "esxiVSwitch0" {
  name = "esxiVSwitch0"
}
```

Bad:

```hcl
resource "vsphere_host_virtual_switch" "esxiVSwitch0" {
  name       = "esxiVSwitch0"
  host       = "unused"
  datacenter = "unused"
}
```